RecoEcal/EgammaCoreTools/src/EcalClusterTools.cc

```diff
@@ -1307,8 +1307,11 @@ float EcalClusterTools::getNormedIX(const DetId& id)
     if( id.subdetId()==EcalEndcap ) ix = EEDetId(id).ix();
     if( id.subdetId()==EcalShashlik ) ix = EKDetId(id).ix();
     if( ix != 0) {
-      int iXNorm  = ix - 50;
-      if(iXNorm<=0) iXNorm--;
+      float iXNorm; 
+      //in shashlik ix and iy run from 1 to 210
+      if( id.subdetId()==EcalShashlik ) iXNorm  = ix - 105.5;
+      else iXNorm = ix - 50.5;
+      //      if(iXNorm<=0) iXNorm--;
```

Comment:
good to fix this geometry bug here for Shashlik, but i am still not sure if it is a right thing to do to change the iXNorm from integer to half-integer.
And you see, it is not only changing Shashlik, it also changes the PhaseI Endcap.  I don't have the feeling how big can be the impact, but anyway, it would be better to keep PhaseI Endcap unchanged.